DEVELOPMENT.md

```diff
@@ -0,0 +1,91 @@
+# Development
+
+This doc explains how to setup a development environment so you can get started
+contributing to `Knative-GCP`.
+Also take a look at:
+
+- [The pull request workflow](https://www.knative.dev/contributing/contributing/#pull-requests)
+- [How to add and run e2e tests](./test/e2e/README.md)
+- [Iterating](#iterating)
+- [Format check](#format-check)
+
+## Getting started
+
+1. [Create and checkout a repo fork](#checkout-your-fork)
+
+1. [Installing Knative-GCP](./README.md/#installing-knative-GCP)
+
+### Checkout your fork
+
+The Go tools require that you clone the repository to the
+`src/github.com/google/knative-gcp` directory in your
+[`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
+
+To check out this repository:
+
+1. Create your own
+   [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine:
+
+```shell
+mkdir -p ${GOPATH}/src/github.com/google
+cd ${GOPATH}/src/github.com/google
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/knative-gcp.git
+cd knative-gcp
+git remote add upstream https://github.com/google/knative-gcp.git
+git remote set-url --push upstream no_push
+```
+
+_Adding the `upstream` remote sets you up nicely for regularly
+[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+
+Once you reach this point you are ready to do a full build and deploy as
+follows.
+
+## Iterating
+
+As you make changes to the code-base, there are two special cases to be aware
+of:
+
+- **If you change a type definition ([pkg/apis/](./pkg/apis/.)),** then you must
+  run [`./hack/update-codegen.sh`](./hack/update-codegen.sh).
+- **If you change a package's deps** (including adding external dep), then you
+  must run [`./hack/update-deps.sh`](./hack/update-deps.sh).
```

Comment:
We used to have ` ./hack/../vendor/knative.dev/test-infra/scripts/library.sh: line 29: conditional binary operator expected` error when running `./hack/update-codegen.sh` in Mac, the root cause is bash version in Mac needs to upgrade (the default version for bash is dated from 2007...). Might worth to indicate this here.

Reply:
done